### PR TITLE
Support for JWT

### PR DESCRIPTION
**Pull request by sorenmat**
* Did  a bit of clean up of package names.
* Introduced an interface for the services to implmenent.
This is done so we can update the interface definition when adding new
rest methods, and that will give us an not implemented error in the
actual handler with the business code
* Moved generated code from pkg/generated to generated

### CODE REVIEW

File: templates/kapeta/block-type-service/auth-middleware.hbs [hunk 1]
```diff
@@ -1,48 +1,19 @@
...
-    }));
-
-    {{#provides 'kapeta/resource-type-auth-jwt-provider'}}
-    router.use(jwksRoutes(providerKeyStore));
-    {{/provides}}
-
-    return router;
-};
-*/
+{{#consumes 'kapeta/resource-type-auth-jwt-consumer'}}
```
Comment (hofmeister) [inline templates/kapeta/block-type-service/auth-middleware.hbs:12]: Should  the middleware not also be added if its a provider ?

File: templates/kapeta/block-type-service/main.go.hbs [hunk 2]
```diff
@@ -28,9 +31,12 @@ func main() {
     if err != nil {
         panic(err)
     }
+    {{#consumers-of-type 'kapeta/resource-type-auth-jwt-consumer'}}
```
Comment (hofmeister) [inline templates/kapeta/block-type-service/main.go.hbs:34]: Middleware for providers ?

File: test/resources/examples/users/generated/services/users_interface.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+package services
+
+import (
+	"github.com/labstack/echo/v4"
+)
+
+// UsersInterface is an interface for UsersHandler
+// It is used to define the methods that are implemented in the UsersHandler
+type UsersInterface interface {
+	CreateUser(ctx echo.Context) error
```
Comment (hofmeister) [inline test/resources/examples/users/generated/services/users_interface.go:12]: Any way we can make the request body, path params and query params more type safe?

**Review by hofmeister — CHANGES_REQUESTED** [review]

File: test/resources/examples/users/generated/services/users_interface.go (hunk 3 above)
Comment (sorenmat) [reply test/resources/examples/users/generated/services/users_interface.go:12]: Great idea... Yes, will fix

**Review by hofmeister — APPROVED** [review]
Looking good - just minor naming comment

**Pull request merged**